Intro_markdown.Rmd

```diff
@@ -181,30 +181,68 @@ Your new directory will be automatically set as the working directory. If you ha
 
-This runs the following command:
+Assuming that you are working with R on the AP, you are encouraged to use the
+package `renv` to manage your packages.
+
+To synchronise your working R environment with this course's repo, run
 
 ```{r eval = FALSE}
-install.packages("tidyverse")
+renv::restore()
 ```
+If you have no installed `renv` yet, you can do this with
 
-Note that if you are using R on the Analytical Platform the tidyverse package may already be installed, hence the above step can be skipped. 
+```{r eval = FALSE}
+install.packages('renv')
+```
 
-Once a package is installed, you should be able to see it in the packages tab. If you want to use it, you can load it by ticking the appropriate box in the packages window. You can also load packages using the library command, which you can put inside your script, so they will automatically load when you run it:
+If prompted to update, enter Y for Yes. This should now update your project 
+environment.
 
-```{r results=FALSE, warning=FALSE, comment=FALSE}
-library("tidyverse")
+This is the command to install/update a package using `renv`:
+
+```{r eval = FALSE}
+renv::install('tidyverse')
 ```
 
+Run this, and the tidyverse package (actually a suite of packages) should update
+to a more recent version.
+
 The package suite `tidyverse` contains many useful packages such as `dplyr` which is a particularly useful package for manipulating and processing data. Many of the functions in the rest of this introductory training are from this package.
 
+There are different ways of using packages  It is possible to load them with library calls, usually found at the start of scripts, e.g.
+
+```{r results=FALSE, warning=FALSE, comment=FALSE}
+library("magrittr")
+```
+
+You can also load packages via the GUI. You should be able to see them in the packages tab. You can load them by ticking the appropriate box in the packages window.
+
+However, for the most part it is considered best practice to not rely on having libraries loaded, but instead to specify them when they are being called, e.g.
+
+```{r results=FALSE, warning=FALSE, comment=FALSE}
+lubridate::days_in_month(Sys.Date())
+```
+
+Here we're using function `days_in_month()` from the `lubridate` package. We're passing to it the current date.
+
+Functions that are 'base' R, or part of a core package such as the `stats` package, which we use below, do not require.
+
+Although including the library call makes the code longer, it also makes it more readable, as anyone who is unfamiliar with the code can see very easily where functions are being called, and which packages are being used. It also avoids the wrong function being called if two functions have the same name and are from different packages. If the package isn't specified by using the double-colon notation then R will use the function from your most recently loaded package and will warn you when you load a package if there is some overlap.
```

Comment:
Replace "and will warn you when you load a package if there is some overlap" with a new sentence something like "You will be warned about any overlaps when you load a package"

Reply:
done